src/Entity/Pap/BuildingBlock.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace App\Entity\Pap;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use App\Entity\Adherent;
+use App\Entity\EntityTimestampableTrait;
+use App\Pap\BuildingStatusEnum;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="pap_building_block")
+ *
+ * @ApiResource(
+ *     attributes={
+ *         "normalization_context": {
+ *             "groups": {"pap_building_block_list"},
+ *             "iri": true,
+ *         },
+ *         "pagination_enabled": false,
+ *     },
+ *     collectionOperations={},
+ *     itemOperations={},
+ * )
+ */
+class BuildingBlock
+{
+    use EntityTimestampableTrait;
+
+    /**
+     * @ORM\Column(type="integer", options={"unsigned": true})
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     */
+    private ?int $id;
+
+    /**
+     * @ORM\Column
+     *
+     * @Groups({"pap_building_block_list"})
+     */
+    private string $name;
+
+    /**
+     * @ORM\Column(length=10)
+     *
+     * @Assert\Choice(
+     *     callback={"App\Pap\BuildingStatusEnum", "toArray"},
+     *     strict=true
+     * )
+     *
+     * @Groups({"pap_building_block_list"})
+     */
+    private ?string $status;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="App\Entity\Pap\Building", inversedBy="buildingBlocks")
+     * @ORM\JoinColumn(nullable=false, onDelete="CASCADE")
+     */
+    private Building $building;
+
+    /**
+     * @var Floor[]|Collection
+     *
+     * @ORM\OneToMany(
+     *     targetEntity="App\Entity\Pap\Floor",
+     *     mappedBy="buildingBlock",
+     *     cascade={"all"},
+     *     orphanRemoval=true,
+     *     fetch="EAGER"
+     * )
+     * @ORM\OrderBy({"number": "ASC"})
+     *
+     * @Groups({"pap_building_block_list"})
+     */
+    private Collection $floors;
+
+    /**
+     * @Gedmo\Blameable(on="create")
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Adherent")
+     * @ORM\JoinColumn(onDelete="SET NULL")
+     */
+    private ?Adherent $createdBy;
```

Comment:
`private ?Adherent $createdBy = null;`

otherwise you will have a fatal error if you call $buildingBlock->getCreatedBy(); on an object with no creator